rules/src/main/resources/io/github/belgif/rest/guide/validator/rules/oas/Rule-ScopesAreDefinedInSecurityScheme.drl

```diff
@@ -0,0 +1,50 @@
+package io.github.belgif.rest.guide.validator.rules.oas;
+
+import java.util.Map
+import java.util.Set
+
+import org.eclipse.microprofile.openapi.models.security.SecurityScheme;
+import org.eclipse.microprofile.openapi.models.security.SecurityScheme.Type;
+import org.eclipse.microprofile.openapi.models.security.OAuthFlows;
+import org.eclipse.microprofile.openapi.models.security.OAuthFlow;
+import org.eclipse.microprofile.openapi.models.security.SecurityRequirement;
+
+import io.github.belgif.rest.guide.validator.core.ViolationReport;
+import io.github.belgif.rest.guide.validator.core.ApiFunctions;
+import io.github.belgif.rest.guide.validator.core.model.SecurityRequirementDefinition;
+import io.github.belgif.rest.guide.validator.core.model.SecuritySchemeDefinition
+import java.util.HashSet;
+
+global io.github.belgif.rest.guide.validator.core.ViolationReport oas;
+
+function void violationScopes(ViolationReport oas, SecurityRequirementDefinition securityRequirementDefinition, String scope, String securitySchemePointer){
+    oas.addViolation("[scopes]", "Scopes defined in a security requirement must also be defined in the referenced security scheme.", "Scope: <<" + scope + ">> not found in: " + securitySchemePointer,
```

Comment:
there's no explicit rule yet in the guide, you can use [sec-scopes] (which we could add later to the guide)